packages/astro/src/vite-plugin-jsx/index.ts

```diff
@@ -193,7 +193,7 @@ export default function jsx({ settings, logging }: AstroPluginJSXOptions): Plugi
 
 			const { mode } = viteConfig;
 			// Shortcut: only use Astro renderer for MD and MDX files
-			if (id.includes('.mdx') || isMarkdownFile(id, { criteria: 'includes' })) {
+			if (id.includes('.mdx') || isMarkdownFile(id, { criteria: 'endsWith' })) {
```

Comment:
If this causes memory leaks for markdown files, doesn't it happen for `.mdx` files too?

Reply:
Oh good point. Gonna fix.